logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/MLSWelcomeEventHandler.kt

```diff
@@ -80,11 +84,24 @@ internal class MLSWelcomeEventHandlerImpl(
                 resolveConversationIfOneOnOne(event.conversationId)
             }
             .flatMapLeft {
-                if (it is MLSFailure.ConversationAlreadyExists) {
-                    kaliumLogger.w("$TAG: Discarding welcome since the conversation already exists")
-                    Either.Right(Unit)
-                } else {
-                    Either.Left(it)
+                when (it) {
+                    is MLSFailure.ConversationAlreadyExists -> {
+                        kaliumLogger.w("$TAG: Discarding welcome since the conversation already exists")
+                        Either.Right(Unit)
+                    }
+
+                    is MLSFailure.OrphanWelcome -> {
+                        kaliumLogger.w("$TAG: Discarding welcome and joining existing conversation by external commit")
+                        joinExistingMLSConversation(
+                            conversationId = event.conversationId,
+                        ).map {
+                            kaliumLogger.d("$TAG: Successfully joined existing MLS conversation")
+                        }
```

Comment:
can you please add a test case to check if joinExistingMLSConversation is called